xbmc/epg/EpgInfoTag.cpp

```diff
@@ -488,6 +512,83 @@ std::string CEpgInfoTag::Plot(bool bOverrideParental /* = false */) const
   return retVal;
 }
 
+void CEpgInfoTag::SetOriginalTitle(const std::string &strOriginalTitle)
+{
+  CSingleLock lock(m_critSection);
+  m_bChanged |= (m_strOriginalTitle != strOriginalTitle);
+  m_strOriginalTitle = strOriginalTitle;
+}
+
+std::string CEpgInfoTag::OriginalTitle(bool bOverrideParental /* = false */) const
+{
+  std::string retVal;
+  CSingleLock lock(m_critSection);
+  if (bOverrideParental || !m_pvrChannel || !g_PVRManager.IsParentalLocked(*m_pvrChannel))
+    retVal = m_strOriginalTitle;
+  
+  return retVal;
+}
+
+void CEpgInfoTag::SetCast(const std::string &strCast)
+{
+  CSingleLock lock(m_critSection);
+  m_bChanged |= (m_strCast != strCast);
+  m_strCast = strCast;
+}
+
+std::string CEpgInfoTag::Cast(void) const
+{
+  return m_strCast;
```

Comment:
Are you absolutely sure that we don't change existing tags? Yesterday we were at the point that we need mutex protection for CEpgInfoTag...